### ISSUE TITLE
Batch API: Does not allow read jobs for public datasets without API key

### ISSUE THREAD

**Issue by andrewbt** [CartoDB/CartoDB-SQL-API#334]
The Batch API Docs say:

For access to all private tables, and for write access to public tables, an API Key is required to authenticate your queries with the Batch API. The following error message appears if you are using private tables and are not authenticated:

And yet, a POST request to https://athompson.carto.com/api/v2/sql/job with
{ "query" : "select * from railroad_accidents limit 10" }
without API key gets a response

{
"error": "permission denied"
}

cc @rohshar @danicarrion for Python SDK work

**Comment by dgaubert** [comment]
Well, I think we haven't documented it well enough. When you create a new job you are creating a new resource and always you'll need the API key, in spite of the query is a `select`.

By the way, I have a question: Does it make sense to create a _read job_? I mean, there is no way to fetch results from a job that has a `select` using the Batch API. To fetch them we need to wrap that query into a `create table new_table as ...` and then (once the job is done) requesting the new table using SQL API.

I'm going to create a new issue to improve our documentation.

Thanks.

**Comment by rochoa** [comment]
Should we close this one?

**Comment by dgaubert** [comment]
Shouldn't we?
